src/nb_type.cpp

```diff
@@ -810,11 +810,20 @@ NB_NOINLINE char *extract_name(const char *cmd, const char *prefix, const char *
           cmd, s, prefix);
     p += prefix_len;
 
-    // Find the opening parenthesis
+    // Find the opening parenthesis or bracket
+#if PY_VERSION_HEX >= 0x030C0000
+    const char *p2 = strchr(p, '(');
+    const char *p3 = strchr(p, '[');
+    p2 = p2 == nullptr ? p3 : (p3 == nullptr ? p2 : (p2 < p3 ? p2 : p3));
```

Comment:
I think it's fine to have just one version of this code.

Can you write the ternary expression using an `if` statement? It's a little hard to read.